### PR TITLE
Javascript General Coding Guidelines

### PR DESCRIPTION
**Pull request by alex40724**
This PR is a follow up of https://github.com/ILIAS-eLearning/ILIAS/pull/2445, both are currently WIP.

It adresses the first identified JS Coding Requirement, which is to agree on a JS Coding Style. Please add your comments and thoughts in the PR. You are also invited to join the ILIAS Offline Group [1].

I will announce this PR on JF 17th Jan 2020 and ask to make a final decision on JF 9th Mar 2020.

[1] https://docu.ilias.de/goto_docu_grp_8190.html

### CODE REVIEW

File: docs/development/js/js-coding-style.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+```
+/* global $, il */
+``` 
+
+This will prevent any "... is not defined" errors in the StandardJS syntax checker (see next chapter).
+
+## StandardJS
+
+Javascript code should be written using [StandardJS](https://standardjs.com/rules.html) as coding style.
```
Comment (mjansenDatabay) [inline docs/development/js/js-coding-style.md]: Maybe the [Airbnb Coding Standard](https://github.com/airbnb/javascript) could be an alternative worth discussing. According to my frontend colleagues and friends (in other companies), this is the most hyped and (according to them) most comprehensive JavaScript coding style.

[ESlint](https://eslint.org/) with an additional [configuration](https://www.npmjs.com/package/eslint-config-airbnb) could be used to check the `Airbnb Coding Standard`.

Comment (nhaagen) [reply docs/development/js/js-coding-style.md]: I'm all for the commitment to a single standard!
Please note that this one e.g. requires to write variables in CamelCase, which is directly opposite to the code-style for php in ILIAS.
As I said, just a note...

**Comment by alex40724** [comment]
I have added Airbnb style as an alternative and added some comparison.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 17 FEB 2020: We highly appreciate the suggested guideline suggestion and ask all developers to read this document carefully and add comments to this page in case of different opinions. We want to make a final decision about this document in the next Jour Fixe at March 09, 2020.

**Comment by alex40724** [comment]
My **current personal preference** would be to use **Airbnb**. Mainly due to the use of semicolons, which make code more readable for me. Code looks more similar to our PHP code this way.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 09 MAR 2020 : We highly appreciate this PR and accept it for trunk. We prefer the AirBnB style as JS style for ILIAS – because of the use of semicolons and a certain similarity to the PHP coding style in ILIAS.

**Pull request merged**